internal/controller/appconfigurationprovider_controller.go

```diff
@@ -250,6 +250,22 @@ func (reconciler *AzureAppConfigurationProviderReconciler) Reconcile(ctx context
 
 	/* Create secret when there are secret settings */
 	if processor.RefreshOptions.SecretSettingPopulated {
+		// Verify the existence of the secret which is not owned by the current provider
+		for name := range processor.Settings.SecretSettings {
+			if _, ok := existingSecrets[name]; !ok {
```

Comment:
telling from https://github.com/Azure/AppConfiguration-KubernetesProvider/blob/main/internal/controller/processor.go#L229: 
```
processor.Settings.SecretSettings = existingSecrets
```
Does this mean `processor.Settings.SecretSettings` and `existingSecrets` reference to the same object?

Reply:
Correct, it won't be a problem

Reply:
if they reference to the same object, the check here seems to be unnecessary.  when will `if _, ok := existingSecrets[name];` returns a false value?

Reply:
`existingSecrets` is caculated everytime of reconciliation. The `resolvedSecrets` is resolved from `secretReferencesToSolve`.  In normal situations, the keys of them should be the same, we don't really need this check. But we can't 100% sure that the `existingSecrets[secretName]` exists because the customer may accidently remove that secret, although we have correction mechanism to re-generate the removed secret, but there's may has a timing that the key of these two object is not match. Add this judge to safe guard not crashing the process

Reply:
`existingSecrets` is  calculated eveytime of reconciliation

Reply:
Got it, thanks